src/hotspot/share/gc/shenandoah/heuristics/shenandoahYoungHeuristics.cpp

```diff
@@ -135,8 +135,10 @@ bool ShenandoahYoungHeuristics::should_start_gc() {
   // be done, we start up the young-gen GC threads so they can do some of this old-gen work.  As implemented, promotion
   // gets priority over old-gen marking.
   ShenandoahHeap* heap = ShenandoahHeap::heap();
+
+  size_t promo_expedite_threshold = (heap->young_generation()->max_capacity() * ShenandoahEvacReserve) / 512;
```

Comment:
What's the reasoning behind `512` divisor here? Since `ShenandoahEvacReserve` is percents of young/total heap size, this makes `promo_expedite_threshold` about 20% of heap size. Is that what we want?

Reply:
Err, miscomputed. That would be only `1%` of heap size.

Reply:
Yes.  20% of 5%.  The idea: In the rare scenario that we anticipate that promotion will consume over 20% of the evacuation reserve, we'll expedite the cycle so as to improve the likelihood that GC will replenish young allocation pool before it becomes exhausted.